### ISSUE TITLE
Ruby 2.4.1 support

### ISSUE THREAD

**Issue by draconiandev** [brainspec/enumerize#273]
I was testing the gem with a fresh Rails 5.1.0rc1 app. Whenever I try to save a model, it just rolls back and exits me out of console without any indication.

This didn't happen in the beta release of the rails though.

```ruby
# frozen_string_literal: true

class Audience
  belongs_to :client
  extend Enumerize

  enumerize :type, in: %i[T1 T2 T3]
end
```

```ruby
[1] pry(main)> Audience.create!(name: 'Some Audience', start_age: 18, end_age: 90, visitor_type: 'T2', min_spend_value: 150, client_id: 1)
  Client Load (1.4ms)  SELECT  "clients".* FROM "clients" ORDER BY "clients"."id" ASC LIMIT $1  [["LIMIT", 1]]
   (0.2ms)  BEGIN
  SQL (3.5ms)  INSERT INTO "audiences" ("name", "visitor_type", "start_age", "end_age", "min_spend_value", "client_id", "created_at", "updated_at") VALUES ($1, $2, $3, $4, $5, $6, $7, $8) RETURNING "id"  [["name", "Fourth Audience"], ["visitor_type", "T2"], ["start_age", 18], ["end_age", 90], ["min_spend_value", 150], ["client_id", 1], ["created_at", "2017-04-06 06:16:12.072160"], ["updated_at", "2017-04-06 06:16:12.072160"]]
   (0.3ms)  ROLLBACK
```

Has it anything to do with the magic frozen string literal comment?

**Comment by draconiandev** [comment]
Yup! It is because of the magic comment and ruby 2.4.1. This behaviour is not seen in ruby 2.4.0 or 2.3.3. 
If you plan to use 2.4.1 with rails 5.1, you need to take out the comment for the particular model or use other versions of ruby as of now.

**Comment by lest** [comment]
I wasn't able to reproduce the issue. Also, I updated tests to include Ruby 2.4.1 and Rails 5.1 in #274.

Can you please push a test application reproducing the issue?

**Comment by draconiandev** [comment]
I will do it on the weekend. Thanks for the response.

**Comment by draconiandev** [comment]
Here is the [repo](https://github.com/draconiandev/enumerize_issue273_reproduce). The issue is not with either Ruby or Rails per say. But with a particular gem called [meta request](https://rubygems.org/gems/meta_request) when used with enumerize. If you just take that gem out, the problem disappears. However, I used both without a problem when I tried Rails 5.1.0beta1 and Ruby 2.4.0 earlier.

I have no idea what to make of this. All I can see is `SystemStackError: stack level too deep` in the console.
